src/cryptography/hazmat/backends/openssl/x509.py

```diff
@@ -151,3 +156,66 @@ def signature_hash_algorithm(self):
             raise UnsupportedAlgorithm(
                 "Signature algorithm OID:{0} not recognized".format(oid)
             )
+
+    @property
+    def extensions(self):
+        # TODO: remove this when we finish landing support for extensions.
+        warnings.warn(
+            "Extension support is not fully implemented. Unsupported "
+            "extensions are IGNORED at this time."
+        )
```

Comment:
So let's talk about critical. Does critical mean we should error out here if there's something we can't parse, or does it mean the application level code that accesses extensions should error out?